### DISCUSSION TITLE
Realisation on another platform

### DISCUSSION THREAD

**Discussion by gev** [roc-streaming/roc-toolkit#496] [Discuss]
Is it possibble to implement this protocol for PIC 32 Mz MCU 512K RAM 192MHz and mix 4-8 streams?

**Comment by gavv** [comment]
Hi, sorry, seems this message got lost. I understand it's likely not relevant already, but I'll answer anyway.

Theoretically, I think roc is lightweight and modular enough to make such port possible. In practice, such port will require quite a lot of work of course.

Code base uses [target directories](https://roc-streaming.org/toolkit/docs/internals/code_structure.html#id5) to isolate platform-specific parts. For example, network layer is isolated and can be substituted with platform-specific implementation.

Some features are too heavy though. For example, I think we'll need to exclude resampler (we can move it to sender side for example) and LDPC (not sure about Reed-Solomon and OpenFEC in general; though OpenFEC is not the only Reed-Solomon implementation).

Personally I would be very interested to see such porting efforts and would be happy to help.

**Comment by gavv** [comment]
@baranovmv thoughts?
